go-controller/pkg/ovn/gress_policy.go

```diff
@@ -59,15 +59,35 @@ func (pp *portPolicy) getL4Match() (string, error) {
 		}
 	}
 	if len(foundProtocol) == 0 {
-		return "", fmt.Errorf("unknown port protocol %v", pp.protocol)
+		return fmt.Errorf("unknown port protocol %v", pp.protocol)
+	}
+	// We are already configured to match on all ports, so not need
+	// to look for ports.
+	if l4Matches[foundProtocol] == nil {
+		l4Matches[foundProtocol] = make(map[string]string)
+	}
+	if l4Matches[foundProtocol]["portlist"] == "all" {
+		return nil
 	}
 	if pp.endPort != 0 && pp.endPort != pp.port {
-		return fmt.Sprintf("%s && %d<=%s.dst<=%d", foundProtocol, pp.port, foundProtocol, pp.endPort), nil
-
+		if l4Matches[foundProtocol]["portrange"] == "" {
+			l4Matches[foundProtocol]["portrange"] = fmt.Sprintf("%d<=%s.dst<=%d", pp.port,
+				foundProtocol, pp.endPort)
+		} else {
+			l4Matches[foundProtocol]["portrange"] = fmt.Sprintf("%s || %d<=%s.dst<=%d",
```

Comment:
If there are overlapping between port ranges, are they merged already before calling this function?

Reply:
Sounds good. It can be a separate PR (if needed).

Reply:
Yes, LGTM. Thanks @venuiyer